src/java.base/share/classes/java/net/ServerSocket.java

```diff
@@ -799,7 +801,7 @@ public synchronized void setSoTimeout(int timeout) throws SocketException {
      * @since   1.1
      * @see #setSoTimeout(int)
      */
-    public synchronized int getSoTimeout() throws IOException {
+    public int getSoTimeout() throws IOException {
```

Comment:
Is it safe to drop `synchronized` here? Any subclasses/implementations rely on mutual exclusion here?

Reply:
It would be relying on undocumented and inconsistent behavior. I think they are left over from when the getter/setter methods were synchronized with close or the old socket implementation where the Socket and SocketImpl implementation were very tangled. Methods such as xxxReuseAddress don't synchronize, another inconsistency. I can drop the removal of synchronized from the patch but it really don't serve any purpose here.

Reply:
All right then.